src/main/java/seedu/address/logic/commands/UntagCommand.java

```diff
@@ -76,16 +76,22 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_TAG_NOT_FOUND_IN_CONTACT);
         }
 
-        if (!updatedTags.containsAll(tagsToRemove)) {
+        if (tagsToRemove.isEmpty()) {
             throw new CommandException(MESSAGE_TAG_NOT_FOUND_IN_CONTACT);
         }
 
-        updatedTags.removeAll(tagsToRemove);
+        if (!updatedTags.containsAll(tagsToRemove)) {
+            throw new CommandException(MESSAGE_TAG_NOT_FOUND_IN_CONTACT);
+        }
 
-        for (Tag tag : tagsToRemove) {
-            tag.decreaseTaggedCount();
+        for (Tag tag : updatedTags) {
+            if (tagsToRemove.contains(tag)) {
```

Comment:
Is there a reason why you loop through the updatedTags here instead of just looping through the tagsToRemove? If the tags to remove is shorter, might be a bit more efficient